interfaces/builtin/acrn.go

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 Intel Corporation, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package builtin
+
+const acrnSummary = `allows access to the ACRN device`
+
+const acrnBaseDeclarationSlots = `
+  acrn:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const acrnConnectedPlugAppArmor = `
+# Description: Allow access to resources required by ACRN.
+#   allow offline CPU cores
+/sys/devices/system/cpu/cpu[0-9]*/online w,
+
+#   allow write access to ACRN Virtio and Hypervisor service Module
+/dev/acrn_vhm rw,
```

Comment:
Are there any special modules needed for this device? In which case, they may be a need to override KModConnectedPlug() or set connectedPlugKModModules

Reply:
@bboozzoo is there wiki pages documenting these callbacks? I learn it from the `kvm.go`, it looks to me the [KModConnectedPlug()](https://github.com/snapcore/snapd/blob/master/interfaces/builtin/kvm.go#L78-L101) is used to check the kernel module presence depending on the CPU type. But in our case, the `/dev/acrn_vhm` is registered by a [static linked](https://github.com/projectacrn/acrn-kernel/blob/master/kernel_config_uefi_sos#L771) kernel driver, do we need to provide an empty callback? Thanks.

Reply:
Is it possible to build it as a dynamically loaded module? Does it have aliases set up such that it will get automatically loaded?

Reply:
@bboozzoo , the ACRN VHM driver does not support kernel module dynamic loading per this [Kconfig](https://github.com/projectacrn/acrn-kernel/blob/master/drivers/vhm/Kconfig). Any security concern? @gvancuts may provide additional clarification.

Reply:
/cc @shuox to help answer questions about the ACRN kernel driver. The version we use can only be statically compiled in the kernel. The version that was accepted upstreamed more recently (see [this blog](https://projectacrn.org/acrn-hypervisor-service-module-upstreamed-to-linux-kernel/) for additional details) can be configured as a module, but we are are not using this yet with the user-space components from https://github.com/projectacrn/acrn-hypervisor.

Reply:
Yes. As @gvancuts mentioned, VHM driver will be drop (and it had never be upstreamed). The new driver named HSM (derived from VHM) has been accepted by Linux 5.12

The HSM use '/dev/acrn_hsm' as node. And it can be compiled as a module.

Reply:
Is there any reason to not also include `/dev/acrn_hsm` in the AppArmor snippet for this interface and mention the module name in the connectedPlugKModModules as well? AIUI, for your deployment you will have a statically linked kernel module, but going forward in the future others may wish to use this and may be using the dynamically loaded module in 5.12.

Reply:
@anonymouse64 . Thank you for feedbacks. I will include the new device node in the apparmor snippet and add the `connectedPlugKModModules()` in the amended commit.